### PR TITLE
EE-2374 - 3 New GitHub Actions

### PR DESCRIPTION
**Pull request by M-casado**
## Ticket reference
EE-2374

## Overall changes
- [ ] Added GitHub action to validate JSON documents against schemas
- [ ] Added GitHub action to check all files with linters
- [ ] Added GitHub action to create markdown documentation automatically
- [ ] Added new documentation (with diagrams of the workflows)
- [ ] Updated existing documentation

## Future TO-DOs
- [x] Amend documentation, JSONs and code based on the outcome of the linters and JSON validators

### CODE REVIEW

**Comment by M-casado** [comment]
For the **reviewers**: I have checked that the actions are executed as intended in a [test repository](https://github.com/M-casado/tests/pulls?q=is%3Apr++) of my own. I have yet to amend any issue that these checks identify: those fixes will come in the following commits. 

Bare in mind (check the new diagrams) that **the main GH action**, the one that creates the markdown, is **triggered upon the PR being merged**. In other words, it won't appear until this PR is accepted and pushed to ``main``.

**Comment by M-casado** [comment]
@afoix & @andresfsilva The PR is ready for review. The main steps are in the comment above, and to review I would **mainly focus on the GH actions themselves and documentation** I created for it. Since I separated those in **different commits** I would check everything up to commit 426fe9a9b205143f7438b869d0ae69cd97f915fd, from which I simply merged from other branches and amended the errors/warnings detected by the checkers.

Things to notice:

- [ ] There seems to be an issue regarding the diagrams I created as documentation (in ``./docs/gh_workflows``). They seemed to render correctly in any web browser I used, but apparently some miniatures are not being rendered in the markdown. I'll check that tomorrow, probably upload them as PNGs for now until it gets fixed. 
- [ ] Do bear in mind [this comment](https://github.com/EbiEga/ega-metadata-schema/pull/42#issuecomment-1102904517). I have added lots of changes in the schemas, as you can see in the later fixes. These will appear in the markdowns after we merge this branch with ``main``, serving also as the test for the last github action. 
- [ ] In the super-linter GH action I disabled (turned to false in ``.github/workflows/super_linter.yml``)  natural language and markdown checks: the errors they were throwing out don't seem relevant to me (e.g. ``Incorrect usage of the term: “node”, use “Node.js” instead``). You can check some [here](https://github.com/EbiEga/ega-metadata-schema/runs/6097382623?check_suite_focus=true).

File: docs/json_browser/README.md [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+(with their details, constraints, etc.)
 
-Hence, take a look at the [markdown's summary](markdowns/README.md). There, you will find the section **Top-level Schemas**, which will list the main metadata objects (e.g. ``ArrayAssay``). If you click on any of these, you will be redirected to their corresponding records, with diverse details (such as ``title`` or ``description``) and, most importantly, a list with all their ``properties`` (with their details, constraints, etc.) 
\ No newline at end of file
+## How they are created
+Markdown summaries constitute the JSON Browser, and are automatically generated though the usage of **GitHub actions** ([``markdown_generation``](../../.github/workflows/markdown_creation.yml)) and the [**jsonschema2md**](https://github.com/adobe/jsonschema2md) tool. Every pull request against the ``main`` branch of this repository will trigger the workflow to automatically generate updated
```
Comment (andresfsilva) [inline docs/json_browser/README.md:16]:
```suggestion
Markdown summaries constitute the JSON Browser, and are automatically generated though the usage of **GitHub actions** ([``markdown_generation``](../../.github/workflows/markdown_creation.yml)) and the [**jsonschema2md**](https://github.com/adobe/jsonschema2md) tool. Every pull request merged into the ``main`` branch of this repository will trigger the workflow to automatically generate updated
```

**Review by andresfsilva — APPROVED** [review]
Nice work! Liked this new automation

**Comment by M-casado** [comment]
There seems to be a problem at Super-linter, its page does not load and thus the action fails to pull it. But it passed filters before the last commit, which was minimal and should not make the changes fail the linter.

**Comment by M-casado** [comment]
Re-run the last job today and it worked (super-linter fixed their issues), so my previous comment can be omitted.

**Review by afoix — APPROVED** [review]
All looks perfect to me @M-casado 😊 AGree with the suggestion from Andres so you can go ahead and merge! Good work!

**Pull request merged**